### DISCUSSION TITLE
latest terser-webpack-plugin with webpack 5

### DISCUSSION THREAD

**Discussion by AdelFattakhova** [webpack/webpack#17017] [Q&A] [answered]
I'm migrating from Webpack 4 to Webpack 5 and faced this issue with terser-webpack-plugin.

With old version of terser-webpack-plugin we used `cache: true` and `sourceMap: false` options. For now, those options were deprecated.

Regarding `sourceMap` I understand that it now considers Webpack's `devtool` option instead. BUT what if for Webpack and terser-webpack-plugin we need different source map settings? How do I set `sourceMap` to `false` for terser-webpack-plugin when I have `devtool` set to something else?

Another question is about `cache`. Before it was deprecated, it was set to `true` by default. Does it still behave the same?

**Comment by alexander-akait** [comment, marked as answer]
> Another question is about cache. Before it was deprecated, it was set to true by default. Does it still behave the same?

Webpack has built-in cache https://webpack.js.org/configuration/cache/#cachetype, if you set `type: 'filesystem',`, you will cache not only terser output, but all webpack build (try it and the next build will be faster), note - when you use `filesystem` memory cache will be used too (it is like layers)

> Regarding sourceMap I understand that it now considers Webpack's devtool option instead. BUT what if for Webpack and terser-webpack-plugin we need different source map settings? How do I set sourceMap to false for terser-webpack-plugin when I have devtool set to something else?

What is `devtool` value for production? It was removed because you can break source maps, that is why terser plugin look at the `devtool` option

**Reply by AdelFattakhova** [reply]
Thank you for such a quick response! I'll try using Webpack's cache settings, thanks.

> What is devtool value for production? It was removed because you can break source maps, that is why terser plugin look at the devtool option

It's `cheap-source-map`

**Reply by alexander-akait** [reply]
But what is sense to use `cheap-source-map` for devtool if you will disable source maps in terser? Maybe I am missing something, can you describe use case?

**Reply by AdelFattakhova** [reply]
Great question! 😅 That's actually something that's been living in our project for years and now I have no idea about the initial intentions :)) It's a mystery how nobody still hasn't noticed that we don't have source maps anywhere...

I'll ask my colleagues (maybe somebody remembers something) and come back here tomorrow

**Reply by AdelFattakhova** [reply]
we resolved the issue, will set `devtool` to the specific value in each environment. thanks for your help 👍🏼
